php/libraries/Database.class.inc

```diff
@@ -108,7 +108,7 @@ class Database
             }
         } else {
             // name the connection
-            $connection = md5($database.$host.$username.$password.$trackChanges);
+            $connection = md5($database.$host.$username.$password);
```

Comment:
why are you removing trackchanges from here? I think if it's removed singleton will return the same object for trackchanges true and false to the same db.

Reply:
phan doesn't like the idea of concatenating a boolean value to a string.

We could change it so that the value appended to the string ends up being explicitly `1` or `0` depending on the value of `$trackChanges` though

Reply:
It's just constructing a string to take an md5 sum of, so maybe we can use string interpolation instead of concatenation to make phan happy?

Reply:
Just tried `$connection = md5("$database$host$username$password$trackChanges");` and the same warning was raised by phan.

> php/libraries/Database.class.inc:111 PhanTypeSuspiciousStringExpression Suspicious type bool of a variable or expression used to build a string. (Expected type to be able to cast to a string)

Reply:
I guess building the string to pass to md5 and conditionally adding some string based on the bool instead of inlining is the only choice, then..